File: app/src/main/java/com/suzei/minote/data/dao/NotesDao.java
package com.suzei.minote.data.dao;

import com.suzei.minote.data.entity.Notes;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface NotesDao {

    @Query("SELECT * FROM NOTES")
    List<Notes> findAllNotes();

    @Query("SELECT * FROM NOTES WHERE _id=:itemId")
    Notes findNoteById(int itemId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveNote(Notes note);

    @Delete
    void deleteNote(Notes note);

}

File: app/src/main/java/com/suzei/minote/data/dao/NotesDao.kt
package com.suzei.minote.data.dao

import com.suzei.minote.data.entity.Notes

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface NotesDao {

    @Query("SELECT * FROM NOTES")
    fun findAllNotes(): List<Notes>

    @Query("SELECT * FROM NOTES WHERE _id=:itemId")
    fun findNoteById(itemId: Int): Notes

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveNote(note: Notes)

    @Delete
    fun deleteNote(note: Notes)

}
